app/prepare_proposal.go

```diff
@@ -25,7 +24,7 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 
 	// build the square from the set of valid and prioritised transactions.
 	// The txs returned are the ones used in the square and block
-	dataSquare, txs, err := square.Build(txs, appconsts.MaxSquareSize)
+	dataSquare, txs, err := square.Build(txs, app.GovMaxSquareSize(sdkCtx))
```

Comment:
wouldn't this have conflicts with: https://github.com/celestiaorg/celestia-app/pull/1768/files#diff-abccb046bddf0559ea4defb5570fe3d6b417f443ddf37c657a7b8f1e485903ea ?